src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1062,6 +1064,52 @@ private bool ImportIntrinsicCall(MethodDesc method)
 
             switch (method.Name)
             {
+                case "InitializeArray":
+                    if (metadataType.Namespace == "System.Runtime.CompilerServices" && metadataType.Name == "RuntimeHelpers")
+                    {
+                        var fieldSlot = (LdTokenEntry<FieldDesc>)_stack.Pop();
+                        var arraySlot = _stack.Pop();
+
+                        var fieldDesc = fieldSlot.LdToken;
+                        var dataBlob = _compilation.GetFieldRvaData(fieldDesc).GetData(_compilation.NodeFactory, false);
+                        Debug.Assert(dataBlob.Relocs.Length == 0);
+                        var memBlock = dataBlob.Data;
+
+                        // TODO: Need to do more for arches with different endianness?
+                        LLVMValueRef globalDataHolder = LLVM.AddGlobal(Module, LLVM.ArrayType(LLVM.Int8Type(), (uint)memBlock.Length), $"array.init.{_arrayInitializerHolderCount++}");
```

Comment:
I think the WebAssemblyObjectWriter should be able to write this. @MichalStrehovsky, might you have pointers on how to ensure it does that?

Reply:
The `_compilation.GetFieldRvaData(fieldDesc)` call just above returns an `ISymbolNode` - it should be possible to just generate a reloc to it (same as when the codegen needs to e.g. load a reference to an EEType to do a cast or an allocation).

Reply:
(Might need to also add it to `_dependencies` so that the data blob is marked as used in the dependency graph.)